pkg/jx/cmd/common_helm.go

```diff
@@ -65,31 +62,26 @@ func (o *CommonOptions) registerLocalHelmRepo(repoName, ns string) error {
 			return errors.Wrapf(err, "failed to remove the repository '%s'", repoName)
 		}
 	}
-	return o.Helm().AddRepo(repoName, helmUrl)
+	return o.Helm().AddRepo(repoName, helmUrl, "", "")
 }
 
 // addHelmRepoIfMissing adds the given helm repo if its not already added
-func (o *CommonOptions) addHelmRepoIfMissing(helmUrl string, repoName string) error {
-	return o.addHelmBinaryRepoIfMissing(helmUrl, repoName)
+func (o *CommonOptions) addHelmRepoIfMissing(helmUrl, repoName, username, password string) error {
+	return o.addHelmBinaryRepoIfMissing(helmUrl, repoName, username, password)
 }
 
-func (o *CommonOptions) addHelmBinaryRepoIfMissing(helmUrl string, repoName string) error {
+func (o *CommonOptions) addHelmBinaryRepoIfMissing(helmUrl, repoName, username, password string) error {
 	missing, err := o.Helm().IsRepoMissing(helmUrl)
 	if err != nil {
 		return errors.Wrapf(err, "failed to check if the repository with URL '%s' is missing", helmUrl)
 	}
 	if missing {
 		log.Infof("Adding missing Helm repo: %s %s\n", util.ColorInfo(repoName), util.ColorInfo(helmUrl))
-		err = o.retry(6, 10*time.Second, func() (err error) {
```

Comment:
I feel this is unnecessary (and will bloat the output in the case of a failure), but it's not a strong issue so if you feel it should still remain in then say here and I'll reinstate it.

At present, if you enter the password incorrectly, it will fail all 6 times:

```
$ jx create addon cloudbees --sso

You will need your username and password to install this addon while it is in preview.
To register to get your username/password to to: https://pages.cloudbees.com/K8s

? CloudBees Preview username myusername
? CloudBees Preview password [? for help] **********
Adding missing Helm repo: cb https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com
retrying after error:failed to add the repository 'cb' with URL 'https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com': failed to run 'helm repo
 add cb https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com' command in directory '', output: 'Error: Looks like "https://myusername:mypassword
@chartmuseum.jx.charts-demo.cloudbees.com" is not a valid chart repository or cannot be reached: Failed to fetch https://myusername:mypassword@chartmuseum.jx.char
ts-demo.cloudbees.com/index.yaml : 401 Unauthorized': exit status 1
retrying after error:failed to add the repository 'cb' with URL 'https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com': failed to run 'helm repo
 add cb https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com' command in directory '', output: 'Error: Looks like "https://myusername:mypassword
@chartmuseum.jx.charts-demo.cloudbees.com" is not a valid chart repository or cannot be reached: Failed to fetch https://myusername:mypassword@chartmuseum.jx.char
ts-demo.cloudbees.com/index.yaml : 401 Unauthorized': exit status 1
retrying after error:failed to add the repository 'cb' with URL 'https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com': failed to run 'helm repo
 add cb https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com' command in directory '', output: 'Error: Looks like "https://myusername:mypassword
@chartmuseum.jx.charts-demo.cloudbees.com" is not a valid chart repository or cannot be reached: Failed to fetch https://myusername:mypassword@chartmuseum.jx.char
ts-demo.cloudbees.com/index.yaml : 401 Unauthorized': exit status 1
retrying after error:failed to add the repository 'cb' with URL 'https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com': failed to run 'helm repo
 add cb https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com' command in directory '', output: 'Error: Looks like "https://myusername:mypassword
@chartmuseum.jx.charts-demo.cloudbees.com" is not a valid chart repository or cannot be reached: Failed to fetch https://myusername:mypassword@chartmuseum.jx.char
ts-demo.cloudbees.com/index.yaml : 401 Unauthorized': exit status 1
retrying after error:failed to add the repository 'cb' with URL 'https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com': failed to run 'helm repo
 add cb https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com' command in directory '', output: 'Error: Looks like "https://myusername:mypassword
@chartmuseum.jx.charts-demo.cloudbees.com" is not a valid chart repository or cannot be reached: Failed to fetch https://myusername:mypassword@chartmuseum.jx.char
ts-demo.cloudbees.com/index.yaml : 401 Unauthorized': exit status 1
error: after 6 attempts, last error: failed to add the repository 'cb' with URL 'https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com': failed t
o run 'helm repo add cb https://myusername:mypassword@chartmuseum.jx.charts-demo.cloudbees.com' command in directory '', output: 'Error: Looks like "https://myuse
rname:mypassword@chartmuseum.jx.charts-demo.cloudbees.com" is not a valid chart repository or cannot be reached: Failed to fetch https://myusername:mypassword@cha
rtmuseum.jx.charts-demo.cloudbees.com/index.yaml : 401 Unauthorized': exit status 1

```

Code originally committed here: https://github.com/jenkins-x/jx/commit/42110272b81b455585edd1bb0f5c2a06687df30b